### ISSUE TITLE
WAV files have slight corruption at end

### ISSUE THREAD

**Issue by DaVince** [spheredev/neosphere#35]
For some reason, wav files end with a slight bit of distortion at the end of playback. See [this test game](https://dl.dropboxusercontent.com/u/26026874/sphere/wavtest.zip), where you can play back some WAV files and Ogg conversions.

(Note: the first sample, bleep, also has some very slight corruption at the very end of the sample, but it's much less noticeable.)

I suspect something with Allegro's sound playback.

**Comment by fatcerberus** [comment]
Hm, I think it was mentioned at some point that Allegro's WAV routines were buggy or something, so it's possible.  I'll look into it and file a bug report if necessary.

**Comment by fatcerberus** [comment]
Just for the heck of it, could you test this on the latest master?  The sound loading code has changed as a result of adding SphereFS, so maybe I accidentally fixed it...

**Comment by DaVince** [comment]
Good work with SphereFS! Unfortunately, the code doesn't compile at all right now:

```
gcc -o obj/spherefs.o -c obj/spherefs.c
obj/spherefs.c:345:1: error: conflicting types for 'sfs_fseek'
 sfs_fseek(sfs_file_t* file, long offset, int origin)
 ^
In file included from obj/minisphere.h:31:0,
                 from obj/spherefs.c:1:
obj/spherefs.h:29:13: note: previous declaration of 'sfs_fseek' was here
 bool        sfs_fseek  (sfs_file_t* file, long offset, sfs_seek_t origin);
             ^
scons: *** [obj/spherefs.o] Error 1
scons: building terminated because of errors.
```

**Comment by fatcerberus** [comment]
Oops.  Again, MSVC is dumb and didn't catch that.  Change `int origin` to `sfs_seek_t origin` and it should compile.

**Comment by DaVince** [comment]
Still didn't compile due to forgetting to include particles.c in SConscript. ;)

Anyway, after compiling it, there's no difference; the audio still glitches the same way. I think this is more likely to be an audio buffer problem or something.

**Comment by fatcerberus** [comment]
Probably, yeah.  Maybe even a similar issue to what caused that MOD segfault a while back.

**Comment by fatcerberus** [comment]
Just posted this on the Allegro development forums:
https://www.allegro.cc/forums/thread/615461

So hopefully this will be fixed soon!

**Comment by fatcerberus** [comment]
The cause of the issue has been found:
https://github.com/liballeg/allegro5/issues/474#issuecomment-165702885

This will be fixed in minisphere 2.1.5.
